common/src/main/java/net/whimxiqal/journey/data/DataVersion.java

```diff
@@ -78,59 +86,76 @@ public static DataVersion fromString(String versionString) {
-    File versionFile = Journey.get().proxy().dataFolder().resolve(VERSION_FILE_NAME).toFile();
-    if (!versionFile.exists()) {
-      writeVersion(DataVersion.V000);
-    }
     return readVersion();
   }
 
-  static void writeVersion(DataVersion version) {
+  static DataVersion readVersion() {
     File versionFile = Journey.get().proxy().dataFolder().resolve(VERSION_FILE_NAME).toFile();
-    if (!versionFile.exists()) {
-      boolean created = false;
-      try {
-        created = versionFile.createNewFile();
-      } catch (IOException e) {
-        e.printStackTrace();
+
+    // This should trigger the migration from V1 to V2
+    if (versionFile.exists()) {
+      Journey.logger().info("Version file exists, assuming V1.");
+      versionFile.delete();
+      return DataVersion.V001;
+    }
+
+    switch (Settings.STORAGE_TYPE.getValue()) {
+      case MYSQL -> {
+        MySqlConnectionController mysqlController = new MySqlConnectionController();
+        try (Connection connection = mysqlController.establishConnection()) {
+          PreparedStatement statement = connection.prepareStatement("SHOW TABLES LIKE 'journey_db_version'");
+          ResultSet resultSet = statement.executeQuery();
+
+          if (!resultSet.next()) {
+            Journey.logger().warn("No DB Version table exists! Assuming V0");
+            return DataVersion.V000;
+          }
+
+          PreparedStatement readStatement = connection.prepareStatement("SELECT Max(db_version) as DBVersion FROM journey_db_version;");
+          ResultSet readResultSet = readStatement.executeQuery();
+
+          if (readResultSet.next()) {
+            return DataVersion.fromInt(readResultSet.getInt("DBVersion"));
+          } else return DataVersion.V000;
+        } catch (SQLException e) {
+          e.printStackTrace();
+        }
+      }
+      case SQLITE -> {
+        String sqliteAddress = "jdbc:sqlite:" + Journey.get().proxy().dataFolder() + "/" + DATABASE_FILE_NAME;
+        SqliteConnectionController sqliteController = new SqliteConnectionController(sqliteAddress);
+
+        try (Connection connection = sqliteController.establishConnection()) {
+          PreparedStatement statement = connection.prepareStatement("SELECT name FROM sqlite_master WHERE type='table' AND name='journey_db_version'");
+
+          ResultSet resultSet = statement.executeQuery();
+
+          if (!resultSet.next()) {
+            Journey.logger().warn("No DB Version table exists! Assuming V0");
+            return DataVersion.V000;
+          }
+
+          PreparedStatement readStatement = connection.prepareStatement("SELECT Max(db_version) as DBVersion FROM journey_db_version;");
+          ResultSet readResultSet = readStatement.executeQuery();
+
+          if (readResultSet.next()) {
+            return DataVersion.fromInt(readResultSet.getInt("DBVersion"));
+          } else return DataVersion.V000;
+        } catch (SQLException e) {
+          e.printStackTrace();
+        }
```

Comment:
I don't love all this database-engine specific code buried in a static method on the `DataVersion` type. In fact, I never liked that I put any of this IO stuff in this file. Can we move most of this logic to an accessible place in `DataManager`?

The execution should go something like this:
1. Read current version
    1. if version file exists -> return V001  (add comments about how this file is legacy, and should probaby move this to a helper function with a warning about how the file should never be used in the future)
    1. if `journey_db_version` table does not exist -> no version! **Run schema files and set version to latest**
        1. The schema files are up-to-date, so you can just run those files, no need to run any sort of migrations.
        1. The logic for this check should be abstracted into `Sqlite` and `MySql` handlers deriving from a shared interface, like how all other database-engine-specific code is. That way, this switch case here can be removed and the only database-engine setting check is done during DataManager initialization.
        1. `journey_db_version` should be a static final variable somewhere, probably in `DataManager`, since it's a "magic value"
    1. Otherwise, read `db_version`
1. If any work was done (migrations were run or the schema was created for the first time) and therefore the database version was updated, save the updated version number to the database table


You have most of that sequence here, but the `journey_db_version` not existing case is wrong and the database version should be updated after all is said and done here. And, it's hard to read with the `switch` and it can be more formalized and commented